drivers/xen/events.c

```diff
@@ -219,7 +220,7 @@ static void events_isr(void *data)
 	 */
 	vcpu->evtchn_upcall_pending = 0;
 
-	dmb();
+	barrier_data_memory_fence_full();
```

Comment:
I believe, we need `barrier_data_synchronization_fence_full` in this case, as this part synchronizes access to vCPU structures between Zephyr and Hypervisor. I am not sure that GCC's idea of synchronization is valid for inter-VM communications, so I believe we need proper `dsb sy` there.

Reply:
Previously `dmb()` was expanding to `dmb sy`. Now with this PR, GCC is expanding `barrier_data_memory_fence_full()` to `dmb ish`.

Are you suggesting that using `dmb ish` instead `dmb sy` is the problem here? Or that we should use `dsb` instead of `dmb`? Because in the latter case the problem was present even before this PR so that should be fixed in a different PR possibly (or at least commit).